extensions-contrib/tdigestsketch/src/main/java/org/apache/druid/query/aggregation/tdigestsketch/sql/TDigestSketchQuantileSqlAggregator.java

```diff
@@ -0,0 +1,229 @@
+      final AggregateCall aggregateCall,
+      final Project project,
+      final List<Aggregation> existingAggregations,
+      final boolean finalizeAggregations
+  )
+  {
+    // This is expected to be a tdigest sketch
+    final DruidExpression input = Expressions.toDruidExpression(
+        plannerContext,
+        rowSignature,
+        Expressions.fromFieldAccess(
+            rowSignature,
+            project,
+            aggregateCall.getArgList().get(0)
+        )
+    );
+    if (input == null) {
+      return null;
+    }
+
+    final AggregatorFactory aggregatorFactory;
+    final String histogramName = StringUtils.format("%s:agg", name);
+
+    // this is expected to be quantile fraction
+    final RexNode quantileArg = Expressions.fromFieldAccess(
+        rowSignature,
+        project,
+        aggregateCall.getArgList().get(1)
+    );
+
+    if (!quantileArg.isA(SqlKind.LITERAL)) {
+      // Quantile must be a literal in order to plan.
+      return null;
+    }
+
+    final double quantile = ((Number) RexLiteral.value(quantileArg)).floatValue();
+    Integer compression = null;
+    if (aggregateCall.getArgList().size() > 2) {
+      final RexNode compressionArg = Expressions.fromFieldAccess(
+          rowSignature,
+          project,
+          aggregateCall.getArgList().get(2)
+      );
+      compression = ((Number) RexLiteral.value(compressionArg)).intValue();
+    }
+
+    // Look for existing matching aggregatorFactory.
+    for (final Aggregation existing : existingAggregations) {
+      for (AggregatorFactory factory : existing.getAggregatorFactories()) {
+        if (factory instanceof TDigestSketchAggregatorFactory) {
+          final TDigestSketchAggregatorFactory theFactory = (TDigestSketchAggregatorFactory) factory;
+
+          // Check input for equivalence.
+          final boolean inputMatches;
+          final VirtualColumn virtualInput = existing.getVirtualColumns()
+                                                     .stream()
+                                                     .filter(
+                                                         virtualColumn ->
+                                                             virtualColumn.getOutputName()
+                                                                          .equals(theFactory.getFieldName())
+                                                     )
+                                                     .findFirst()
+                                                     .orElse(null);
+
+          if (virtualInput == null) {
+            inputMatches = input.isDirectColumnAccess()
+                           && input.getDirectColumn().equals(theFactory.getFieldName());
+          } else {
+            inputMatches = ((ExpressionVirtualColumn) virtualInput).getExpression()
+                                                                   .equals(input.getExpression());
+          }
+
+          if (inputMatches) {
```

Comment:
This should check that the compression and quantile parameters match as well

Reply:
Hmm, not sure how to do that. Compression is part of the TDigestSketchAggregatorFactory and quantile isn't. The quantile is used in postAggregator.

Reply:
Ah, it should just check compression